app/models/webhooks/utilities.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'json_schemer'
+require 'uri'
+
+# data structures built up at class load time then frozen.  This is threadsafe.
+module Webhooks
+  module Utilities
+    module ClassMethods
+      def register_webhook(consumer_id, consumer_name, subscription, api_guid)
```

Comment:
Please comment that we are assuming the subscription has been through the validator. To avoid empty subscription arrays